src/Diagnostics/FxCop/Desktop.Analyzers/CSharp/Shared/SyntaxNodeHelper.cs

```diff
@@ -0,0 +1,262 @@
+            if (methodDecl != null)
+            {
+                ParameterListSyntax paramList = methodDecl.ParameterList;
+                if (paramIndex < paramList.Parameters.Count)
+                {
+                    EqualsValueClauseSyntax equalsValueNode = paramList.Parameters[paramIndex].Default;
+                    if (equalsValueNode != null)
+                    {
+                        return equalsValueNode.Value;
+                    }
+                }
+            }
+            return null;
+        }
+
+        protected override IEnumerable<SyntaxNode> GetCallArgumentExpressionNodes(SyntaxNode node, CallKind callKind)
+        {
+            if (node != null)
+            {
+                ArgumentListSyntax argList = null;
+                SyntaxKind kind = node.Kind();
+                if ((kind == SyntaxKind.InvocationExpression) && ((callKind & CallKind.Invocation) != 0))
+                {
+                    var invocationNode = (InvocationExpressionSyntax)node;
+                    argList = invocationNode.ArgumentList;
+                }
+                else if ((kind == SyntaxKind.ObjectCreationExpression) && ((callKind & CallKind.ObjectCreation) != 0))
+                {
+                    var invocationNode = (ObjectCreationExpressionSyntax)node;
+                    argList = invocationNode.ArgumentList;
+                }
+                if (argList != null)
+                {
+                    return argList.Arguments.Select(arg => arg.Expression);
+                }
+            }
+
+            return Enumerable.Empty<SyntaxNode>();
+        }
+
+        public override IEnumerable<SyntaxNode> GetObjectInitializerExpressionNodes(SyntaxNode node)
+        {
+            var empty = Enumerable.Empty<SyntaxNode>();
+            if (node == null)
+            {
+                return empty;
+            }
+
+            SyntaxKind kind = node.Kind();
+            if (kind != SyntaxKind.ObjectCreationExpression)
+            {
+                return empty;
+            }
+
+            var objectCreationNode = (ObjectCreationExpressionSyntax)node;
+            if (objectCreationNode.Initializer == null)
+            {
+                return empty;
+            }
+
+            return objectCreationNode.Initializer.Expressions;
+        }
+
+        public override bool IsMethodInvocationNode(SyntaxNode node)
+        { 
+            if (node == null)
+            {
+                return false;
+            }
+            SyntaxKind kind = node.Kind();
+            return kind == SyntaxKind.InvocationExpression || kind == SyntaxKind.ObjectCreationExpression;
+        }
+
+        public override bool IsObjectCreationExpressionUnderFieldDeclaration(SyntaxNode node)
+        {
+            return node != null &&
+                   node.Kind() == SyntaxKind.ObjectCreationExpression &&
+                   node.AncestorsAndSelf().OfType<FieldDeclarationSyntax>().FirstOrDefault() != null;
```

Comment:
@genlu Same goes here, you can get rid of `IsObjectCreationExpressionUnderFieldDeclaration`. Have the code block start action save off the member symbol and the nested syntax node action just check if owning executable code member is a field symbol.